src/Cli/Utils.cs

```diff
@@ -96,8 +96,13 @@ public static EntityAction[] CreateOperations(string operations, EntityActionPol
                 if (op is EntityActionOperation.All)
                 {
                     HashSet<EntityActionOperation> resolvedOperations = sourceType is EntitySourceType.StoredProcedure ?
-                        EntityAction.ValidStoredProcedurePermissionOperations :
-                        EntityAction.ValidPermissionOperations;
+                        EntityAction.ValidStoredProcedurePermissionOperations : EntityAction.ValidPermissionOperations;
+
+                    if (databaseType is DatabaseType.CosmosDB_NoSQL && resolvedOperations.Contains(EntityActionOperation.Update))
+                    {
+                        resolvedOperations.Add(EntityActionOperation.Patch);
+                    }
+
                     // Expand wildcard to all valid operations (except execute)
```

Comment:
If there are no separate permissions for CosmosDB to accommodate the Patch operation, why do we still have a separate function `ValidatePermissionOperationsForCosmosDB`? Subsequently, if its not needed, there's no need to input database type as an additional argument to the various functions.

Reply:
Sorry, I couldn't find the reference of  `ValidatePermissionOperationsForCosmosDB`.. am I missing something?

Reply:
In AuthorizationResolver, you're 1:1 mapping `Operation.Update` to `Operation.Patch`.
```csharp
// <AuthorizationResolver.cs>
        IEnumerable<EntityActionOperation> validOperation =
            operation is EntityActionOperation.All ? EntityAction.ValidPermissionOperations : new List<EntityActionOperation> { operation };

        //For CosmosDB, add Patch operation to the list of valid operations if Update operation is present.
        if (databaseType is DatabaseType.CosmosDB_NoSQL &&
                    validOperation.Contains(EntityActionOperation.Update))
        {
            List<EntityActionOperation> tempOperationList = validOperation.ToList<EntityActionOperation>();
            tempOperationList.Add(EntityActionOperation.Patch);

            validOperation = tempOperationList;
        };
```

 Back to Ani's comment, it doesn't look like AuthorizationResolver needs to pass DatabaseType==CosmosDB (or pass dbtype at all) and all you need to do is check whether the `operation.Update` permission is allowed.

```csharp
//<CosmosMutationEngine.cs>
 EntityActionOperation.Patch =>
                    _authorizationResolver.AreColumnsAllowedForOperation(entityName,
                        roleName: clientRole,
                        operation: EntityActionOperation.Patch, // --> This could just be EntityActionOperation.Update ?
                        columns: inputArgumentKeys),
```

Reply:
Thanks for the review @Aniruddh25 and Thanks for the explanation @seantleonard  I have made this changes.